### ISSUE TITLE
Bookmarklet no longer works due to CSP

### ISSUE THREAD

**Issue by overshard** [pinry/pinry#96]
https://bugzilla.mozilla.org/show_bug.cgi?id=866522

**Comment by irrg** [comment]
Is there any progress on this? Looks like this has been an issue for close to 3 years?

**Comment by winkidney** [comment]
@irrg Use our chrome / firefox plugin to do what you want : ) https://github.com/winkidney/browser-pinry

**Comment by irrg** [comment]
@winkidney does that require https for the pinry install? Because I can't get it to work and it's htting https://myserver

**Comment by winkidney** [comment]
@irrg It always requires https-install of Pinry on https-target-site (for example https://twittter.com).

Any problem on configure https certificate for your Pinry instance?  On an http target site, http-install of Pinry is ok.

I am currently use Let's-Encrypt's service to serve my https instance of Pinry.

**Comment by winkidney** [comment]
No further response, closed.
Feel free to reopen this issue if you have any further question or problem : )

**Comment by ChillyCider** [comment]
Sorry to necro this. I found a way to force the bookmarklet to avoid CSP issues. My process was this:

1. Hardcode your Pinry URL into bookmarklet.js.
   ```js
   var PINRY_BASE = 'https://pinry.whatever.info';
   function getFormUrl() {
       return PINRY_BASE + '/pin-creation/from-url?url=';
   }
   ```
2. Minify the file using an online minifier.
3. Stick the code inside of javascript:void(   ).
4. Use it all as your bookmark.

Here's the result, feel free to edit PINRY_BASE and use in your bookmarks.

```js
javascript:void((function(){PINRY_BASE = 'https://pinry.example.com'; function closePinry(){var el=document.getElementById('pinry-images');el.parentNode.removeChild(el)}function getFormUrl(){return PINRY_BASE + '/pin-creation/from-url?url=';}function setCSS(el,css){for(var k in css){el.style[k]=css[k]}}function pageView(){var pinryImages=document.createElement('div');pinryImages.id='pinry-images';setCSS(pinryImages,{position:'fixed',display:'block',zIndex:2147483647,background:'rgba(0, 0, 0, 0.7)',paddingTop:'70px',top:0,bottom:0,left:0,right:0,textAlign:'center',overflowX:'hidden',overflowY:'auto'});var pinryBar=document.createElement('div');pinryBar.id='pinry-bar';setCSS(pinryBar,{display:'block',background:'black',padding:'15px',position:'absolute',zIndex:9002,width:'100%',top:0,borderBottom:'1px solid #555',color:'white',textAlign:'center',fontSize:'22px'});pinryBar.textContent='Pinry Bookmarklet';pinryBar.onclick=closePinry;pinryImages.appendChild(pinryBar);var pinrySort=document.createElement('div');setCSS(pinrySort,{display:'block',position:'absolute',top:'15px',right:'1em',cursor:'pointer'});pinrySort.textContent='\u21D5 Size';pinryBar.appendChild(pinrySort);document.body.appendChild(pinryImages);document.onkeyup=function(e){if(e.keyCode==27){closePinry()}};pinrySort.onclick=function(e){e.stopPropagation();Array.prototype.slice.call(pinryImages.children).sort(function(a,b){return b.getAttribute('pinryArea')-a.getAttribute('pinryArea')}).forEach(function(div){pinryImages.appendChild(div)})}}function imageView(imageUrl){var image=document.createElement('div');setCSS(image,{backgroundImage:'url('+imageUrl+')',backgroundPosition:'center center',backgroundRepeat:'no-repeat',backgroundSize:'cover',display:'inline-block',color:'blue',textShadow:'yellow 0px 0px 2px, yellow 0px 0px 3px, yellow 0px 0px 4px',width:'200px',height:'200px',margin:'15px',cursor:'pointer',border:'1px solid #555'});image.onclick=function(){var popUrl=getFormUrl()+encodeURIComponent(imageUrl);popUrl=popUrl+'&referer='+encodeURIComponent(window.location);popUrl=popUrl+'&description='+encodeURIComponent(document.title);window.open(popUrl);closePinry()};document.getElementById('pinry-images').appendChild(image);return image}var images={},reURL=/url[(]"([^"]+)"[)]/;function addImage(img){if(images[img.src]){return}images[img.src]=true;var w=img.naturalWidth,h=img.naturalHeight;if(w>200&&h>200){var i=imageView(img.src);i.textContent=w+'\u00D7'+h;i.setAttribute('pinryArea',w*h)}}function addAllImagesToPageView(){var images=document.getElementsByTagName('img');for(var i=0;i<images.length;i+=1){addImage(images[i])}['body','div','td'].forEach(function(tagName){var tags=document.getElementsByTagName(tagName);for(var i=0;i<tags.length;i+=1){var m=reURL.exec(tags[i].style.backgroundImage);if(m){var img=new Image();img.onload=function(){addImage(this)};img.src=m[1]}}})}pageView();addAllImagesToPageView();})());
```
